Change type [auth] to [silentAuth] and Converting [BotFrameworkAdapter] to [CloudAdapter] - msgext-search-sso-config sample

Harikrishnan-MSFT opened the pull request.

wenytang-ms commented on samples/msgext-search-sso-config/nodejs/bots/teamsMessagingExtensionsSearchAuthConfigBot.js:
```diff
@@ -250,7 +250,7 @@ class TeamsMessagingExtensionsSearchAuthConfigBot extends TeamsActivityHandler {
                 return {
                     composeExtension: {
-                        type: 'silentAuth',
+                        type: 'auth',
```
once there turn `silentAuth` to `auth`, i think there should no return 412 in `onInvokeActivity` line 353

Harikrishnan-MSFT replied:
Fixed

wenytang-ms commented:
another question is, once we use `auth` instead of `silentAuth`, the teams will never user token, so the auth flow is not `SSO`.
or you redesign this sample?

Harikrishnan-MSFT commented:
Hi @wenytang-ms 

Test Case - 
1. Using BotFrameworkAdapter, I updated all place Auth, generated a token,and tested the sample, which is working well in all cases.
2. Converted BotFrameworkAdapter to CloudAdapter I updated all places Auth, but the token is not generated.
3. Converted BotFrameworkAdapter to CloudAdapter I updated all places with silentAuth and generated a token, but tokenIsExchangeable returned a 500 Service Error.

Currently, I am looking into these issues; once done, I will update

Harikrishnan-MSFT commented:
****Build Pipeline -** 
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/6324552986
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/4eb8e54c-f8a2-4bcd-95fc-8672714fc396)**

wenytang-ms commented on samples/msgext-search-sso-config/nodejs/bots/teamsMessagingExtensionsSearchAuthConfigBot.js:
```diff
@@ -48,28 +47,30 @@ class TeamsMessagingExtensionsSearchAuthConfigBot extends TeamsActivityHandler {
             return {
                 composeExtension: {
-                    type: 'auth',
+                    type: 'silentAuth',
```
if you change here, once user first use link unfurling without consent before, there will show no login prompt.

Harikrishnan-MSFT replied:
I have tested its working..
FYI - 
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/2b70ee9d-4fd5-43f0-8cc0-307caf8a2ca0)
**Click login -** 
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/27a591a5-a1e4-4e64-a650-159072270da3)
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/f5b5be1c-0026-4a9d-8211-797b7d234ec5)

wenytang-ms replied:
great! 
but can you share the request and respone in F12? 
once there recieved the 412 response  but show nothing anymore.
I tried your branch code but show no any login prompt. is there any other settings on teams?
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/75360946/fc95680e-445f-4ba9-afad-51c2752317fc)

Harikrishnan-MSFT replied:
**Please check this document and follow all steps. - [Setup for Bot SSO](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-conversation-sso-quickstart/BotSSOSetup.md) and check your bot-oAuth connection, botID, and password.**

wenytang-ms replied:
yes, I follow all the steps, and query in message extension can show the prompt login link.
so i think the link unfurling has something different with query in message extension
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/75360946/036f2d56-8492-4367-aad8-cb3bb99552ad)

wenytang-ms replied:
once change the `silentAuth` to `auth` in `handleTeamsAppBasedLinkQuery`, it works fine
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/75360946/0b408bb8-34d5-4fdb-8d0e-40f07ea589a2)

Harikrishnan-MSFT replied:
Its working **silentAuth and auth**, we are maintaining contestant perforce.

wenytang-ms replied:
Actually, link unfurling not working with `silentAuth`, please try again and show your F12 as same as following picture
![Uploading image.png…]()

Harikrishnan-MSFT commented:
Build Pipeline - 
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/6334756988
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/e52c2caf-f413-4fec-9515-ffa030cbe8f8)

wenytang-ms commented on samples/msgext-search-sso-config/nodejs/bots/teamsMessagingExtensionsSearchAuthConfigBot.js:
```diff
@@ -229,23 +232,25 @@ class TeamsMessagingExtensionsSearchAuthConfigBot extends TeamsActivityHandler {
-                const signInLink = await context.adapter.getSignInLink(
-                    context,
-                    this.connectionName
+                const { signInLink } = await userTokenClient.getSignInResource(
```
Here I tried, but show "unsupported"
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/75360946/37bddac9-a2b3-46f1-8969-562ab21efa38)
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/75360946/d41d9e78-cabc-4f81-a4e3-3a2f8afa5417)

Harikrishnan-MSFT replied:
Here I tried  - 
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/e0b1b21d-201d-465b-8fa2-326cdc84e900)
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/03244380-54b1-4502-b9d2-82bb0a30acfd)
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/b4fbda23-ff72-494d-b9f8-979d8ee708b8)

wenytang-ms replied:
yes, you tried this work because you consent already. 
I met the `unsupported` failure is not consent yet.

Harikrishnan-MSFT replied:
Yes correct @wenytang-ms,. Please give consent and check its working or not.

Global Administrator can grant consent using following link: https://login.microsoftonline.com/common/adminconsent?client_id=<%appId%>

wenytang-ms replied:
So if this compose action requires user consent before, then why do we need to return a signInLink action next?

Harikrishnan-MSFT replied:
The first time you click login, that time signInLink returns the link and opens the URL.

![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/f8796bde-3392-4804-abf1-3461c104dfc3)

Harikrishnan-MSFT replied:
For example, if you create an app registration for a different tenant, you're testing a different tenant. At that time, we need to give the grant administrator consent.

Global Administrator can grant consent using following link: https://login.microsoftonline.com/common/adminconsent?client_id=<%appId%>

![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/c68e2353-6de4-4af9-a7e1-a4ec5a383826)

Harikrishnan-MSFT commented:
**@wenytang-ms - Could you please review and approve this PR.**

wenytang-ms commented:
> **@wenytang-ms - Could you please review and approve this PR.**

In fact, I ran into some problems with the consent scenario with this sample, maybe we can communicate again offline.

wenytang-ms commented:
Thank you so much for your video record!
please go ahead without my approval.

Pawank-MSFT commented on samples/msgext-search-sso-config/nodejs/README.md:
```diff
@@ -98,8 +98,12 @@ Under the root of this sample folder, build and run by commands:
 
-**Note**: If you are facing any issue in your app, please uncomment [this](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/msgext-search-sso-config/nodejs/index.js#L43) line and put your debugger for local debug.
-    
+**Note**: If you are facing any issue in your app, please uncomment [this](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/msgext-search-sso-config/nodejs/index.js#L57) line and put your debugger for local debug.
```
Wrong line, it should be 43

Harikrishnan-MSFT replied:
43 is old line I have updated the index code file, so the current line is 57.
![image](https://github.com/OfficeDev/Microsoft-Teams-Samples/assets/117716554/d6aeba22-dc53-41d0-b3c4-29b229ec7efa)

Pawank-MSFT reviewed (commented):
Please fix small comment, rest looks ok Approving.

Pawank-MSFT reviewed (approved):
looks correct, Approving!

Prithvi-MSFT commented on samples/msgext-search-sso-config/nodejs/README.md:
```diff
@@ -98,8 +98,12 @@ Under the root of this sample folder, build and run by commands:
+
+## Consent Required
+
+ - Global Administrator can grant consent using following link:  [https://login.microsoftonline.com/common/adminconsent?client_id=](https://login.microsoftonline.com/common/adminconsent?client_id=)<%appId%> 
```
Is admin consent required for this app to run ?

Harikrishnan-MSFT replied:
Not Required 

Fixed
